File: agent/schema/model_session.py
# !/usr/bin/env python
# -*- coding: utf-8 -*-
# =====================================================
# @File   ：model_session
# @Date   ：2025/11/29 11:12
# @Author ：leemysw
#
# 2025/11/29 11:12   Create
# 2026/2/25          重构：增加 session_key 路由支持
# =====================================================

"""
会话 Pydantic 模型

[INPUT]: 依赖 pydantic
[OUTPUT]: 对外提供 ASession / UpdateTitleRequest
[POS]: schema 模块的会话模型定义，被 session_store/session_repository 消费
[PROTOCOL]: 变更时更新此头部，然后检查 CLAUDE.md
"""

from datetime import datetime, timezone
from typing import Optional

from pydantic import BaseModel, Field

from agent.config.config import settings


class ASession(BaseModel):
    """会话模型"""
    session_key: str = Field(..., description="结构化路由键")
    agent_id: str = Field(default=settings.DEFAULT_AGENT_ID, description="智能体 ID")
    session_id: Optional[str] = Field(default=None, description="SDK会话ID")
    room_session_id: Optional[str] = Field(default=None, description="Room 运行时会话 ID")
    room_id: Optional[str] = Field(default=None, description="所属 room ID")
    conversation_id: Optional[str] = Field(default=None, description="所属 conversation ID")
    channel_type: str = Field(default="websocket", description="通道类型")
    chat_type: str = Field(default="dm", description="会话类型")
    status: str = Field(default="active", description="会话状态")
    created_at: datetime = Field(
        default_factory=lambda: datetime.now(timezone.utc),
        description="创建时间",
    )
    last_activity: datetime = Field(
        default_factory=lambda: datetime.now(timezone.utc),
        description="最后活动时间",
    )
    title: Optional[str] = Field(default=None, description="会话标题")
    message_count: int = Field(0, description="消息数量")
    options: Optional[dict] = Field(default=None, description="会话元数据")

    model_config = {"from_attributes": True}


class UpdateTitleRequest(BaseModel):
    title: str

File: internal/protocol/model_session.go
package protocol

import "time"

// Session 表示对外暴露的统一会话模型。
type Session struct {
	SessionKey     string         `json:"session_key"`
	AgentID        string         `json:"agent_id"`
	SessionID      *string        `json:"session_id"`
	RoomSessionID  *string        `json:"room_session_id"`
	RoomID         *string        `json:"room_id"`
	ConversationID *string        `json:"conversation_id"`
	ChannelType    string         `json:"channel_type"`
	ChatType       string         `json:"chat_type"`
	Status         string         `json:"status"`
	CreatedAt      time.Time      `json:"created_at"`
	LastActivity   time.Time      `json:"last_activity"`
	Title          string         `json:"title"`
	MessageCount   int            `json:"message_count"`
	Options        map[string]any `json:"options"`
	IsActive       bool           `json:"is_active"`
}
